oa-enterprise/lib/omniauth/strategies/cas/service_ticket_validator.rb

```diff
@@ -38,21 +38,25 @@ def user_info
         # returns nil if given nil
         def parse_user_info(node)
           return nil if node.nil?
-          hash = {}
-          node.children.each do |e|
-            unless e.kind_of?(Nokogiri::XML::Text) ||
-                   e.name == 'cas:proxies' ||
-                   e.name == 'proxies'
-              # There are no child elements
-              if e.element_children.count == 0
-                hash[e.name.sub(/^cas:/, '')] = e.content
-              elsif e.element_children.count
-                hash[e.name.sub(/^cas:/, '')] = [] if hash[e.name.sub(/^cas:/, '')].nil?
-                hash[e.name.sub(/^cas:/, '')].push parse_user_info e
+          {}.tap do |hash|
+            node.children.each do |e|
+              node_name = e.name.sub(/^cas:/, '')
+              unless e.kind_of?(Nokogiri::XML::Text) || node_name == 'proxies'
```

Comment:
Nice cleanup @jbjonesjr! If we wanted to be even nitpickier, maybe use a `next if` here instead of the `unless` block for readability?